xcm/src/v3/multiasset.rs

```diff
@@ -565,20 +573,22 @@ impl From<Vec<MultiAsset>> for MultiAssets {
 				res.push(last);
 			}
 		}
-		Self(res)
+		// We know the vec has a size smaller than the bound since we didn't error out yet.
+		Ok(Self(BoundedVec::truncate_from(res)))
 	}
 }
 
 impl<T: Into<MultiAsset>> From<T> for MultiAssets {
 	fn from(x: T) -> Self {
-		Self(vec![x.into()])
+		let bounded_vec: MultiAssetVec = BoundedVec::truncate_from(vec![x.into()]);
```

Comment:
I think we can just use the `bounded_vec` macro exported by `bounded_collections`.

Reply:
The `bounded_vec` macro panics if the number of elements is larger than the bound, it should only be used in test code.